scenario_player/utils/configuration/base.py

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Mapping
+from typing import Optional, Union
+
+import structlog
+
+from scenario_player.exceptions.config import ConfigurationError
+
+log = structlog.get_logger(__name__)
+
+
+class ConfigMapping(Mapping):
+    def __init__(self, loaded_yaml: Mapping):
+        self.dict = loaded_yaml
+
+    def __getitem__(self, item):
+        return self.dict[item]
+
+    def __iter__(self):
+        return iter(self.dict)
+
+    def __len__(self):
+        return len(self.dict)
+
+    @staticmethod
+    def assert_option(expression, err: Optional[Union[str, Exception]] = None):
```

Comment:
This could be simplified if it would get the exception class to use from a class variable. Then you wouldn't need to override it on every inherited class.